### ISSUE TITLE
Publisher should not be global

### ISSUE THREAD

**Issue by MovGP0** [davezych/shience#12]
i just did some unit tests and noted a problematic behavior: 

because publisher is global, switching the publisher is problematic in multi-threaded application where multiple publishers are required. instead, the user should provide the publisher as needed via dependency injection.

**Comment by MovGP0** [comment]
suggested change: 

```
        public static Science<TResult> New<TResult>([NotNull]string name, [NotNull]IPublisher publisher)
        {
            if (string.IsNullOrWhiteSpace(name))
            {
                throw new ArgumentNullException(nameof(name));
            }

            if (publisher == null)
            {
                throw new ArgumentNullException(nameof(publisher));
            }

            return new Science<TResult>(name, publisher);
        }
```

**Comment by davezych** [comment]
I'm fine with this, as long as a "global" publisher is still supported. My original goal was to be able to set the publisher once and not force the user to instantiate a publisher for each experiment.

**Comment by davezych** [comment]
It looks like you added this overload in #13 . :+1:

**Comment by davezych** [comment]
Resolved in #13. Closing.
